tinygrad/runtime/ops_clang.py

```diff
@@ -9,15 +9,24 @@
   'Darwin': {'cflags':'-lm -fPIC --rtlib=compiler-rt ', 'ext':'dylib', 'exp':''}
 }[platform.system()]
 
+def _compile_to_lib(prg:str):
+  # TODO: is there a way to not write this to disk?
+  dst = f"{tempfile.gettempdir()}/clang_{hashlib.md5(prg.encode('utf-8')).hexdigest()}.{args['ext']}"
+  if not os.path.exists(dst):
+    subprocess.check_output(args=('clang -shared -O2 -Wall -Werror -march=native -x c '+args['cflags']+' - -o '+dst+'.tmp').split(), input=prg.encode('utf-8'))
+    os.rename(dst+'.tmp', dst)
+  return ctypes.CDLL(dst)
+
+if platform.system() == 'Linux':
+  # Reproducer taken from https://github.com/llvm/llvm-project/issues/56204
+  lib = _compile_to_lib('float __attribute__((noinline)) n(float v) { return -v; } int check() { return (__fp16)n(-1.f); }')
```

Comment:
this is currently race conditioning in CI because of `-n=auto`, see #1258

Reply:
That makes sense. It felt like a race but I couldn't spot where it was coming from. Would the best solution be to hold off on this PR until that's fixed then?

Reply:
Do you want to take a shot at fixing it? that would be best outcome. We can mark this as draft for now.

Reply:
Hmm, I'm not sure how we'd get around writing to disk on all platforms since `CDLL` requires a filepath - not a file object - to load the lib. I thought about it a bit and I think a filesystem lock would be enough for this PR, which can always be replaced with an in-memory file later down the road if a x-plat way to do that is found. It looks like `portalocker` is what we want, which also appears to be used by `pytorch`.